pysmt/oracles.py

```diff
@@ -168,10 +169,73 @@ def get_theory(self, formula):
         """Returns the thoery for the formula."""
         return self.walk(formula)
 
-
 # EOC TheoryOracle
 
 
+# Operators for which Args is an FNode
+DEPENDENCIES_SIMPLE_ARGS = (set(op.ALL_TYPES) - \
+                           (set([op.SYMBOL, op.FUNCTION]) | op.QUANTIFIERS | op.CONSTANTS))
+
+class FreeVarsOracle(pysmt.walkers.DagWalker):
```

Comment:
A possible improvement (useful for function models) could be to have a parameter to control if/how function sysmbols are added in the resulting cone.

I see three modes:
* None: meaning that only variables are considered in the cone
* Function Symbols [default and current behavior]: function names are symbols appearing in the cpone, but function applications are discarded
* Function applications: function applications are considered atomic and are retured together with the cones of theit parameters.